tensorflow/compiler/mlir/hlo/tests/lhlo-legalize-tensor-load-op.mlir

```diff
@@ -0,0 +1,34 @@
+// RUN: mlir-hlo-opt -lhlo-legalize-tensor-load-op %s -o - | FileCheck %s
+
+// test: `memref -> memref.tensor_load -> tensor.extract` -> `memref -> memref.load`
+// CHECK-LABEL: forward_extract_op
+func @forward_extract_op(%arg0: memref<?x?xf32>, %arg1: memref<3xindex>) -> memref<?x?x?xf32> {
+  %c0 = constant 0 : index
+  %c1 = constant 1 : index
+  %c2 = constant 2 : index
+  // CHECK-NOT: memref.tensor_load
+  // CHECK-NOT: tensor.extract
+  // CHECK: memref.load
+  // CHECK: memref.load
+  // CHECK: memref.load
```

Comment:
You should also check that the indices are correct.

Reply:
Done!